dns-server/src/lib.rs

```diff
@@ -86,3 +91,80 @@ pub async fn start_servers(
 
     Ok((dns_server, dropshot_server))
 }
+
+/// An in-memory DNS server running on localhost.
+///
+/// Intended to be used for testing only.
+pub struct InMemoryServer {
+    /// Server storage dir
+    pub storage_dir: tempfile::TempDir,
+    /// DNS server
+    pub server: dns_server::ServerHandle,
```

Comment:
suggestion: `dns_server`?  There are so many "server"s in these structs I find it helpful to be explicit that this is the DNS [protocol] server.